src/widgets/keyboard/lv_keyboard.c

```diff
@@ -49,6 +63,104 @@ const lv_obj_class_t lv_keyboard_class = {
     .name = "keyboard",
 };
 
+#if LV_KEYBOARD_PINYIN
```

Comment:
It would be more elegant to have something like:
```c
#if LV_USE_ARABIC_PERSIAN_CHARS 
  #define PEERSIAN_KEY  "أب",
  #define PEERSIAN_CTRL  LV_KEYBOARD_CTRL_BUTTON_FLAGS | 2,
#else
  #define PEERSIAN_KEY  
  #define PEERSIAN_CTRL
#endif
//Same for Chines
//..

static const char * const default_en_kb_map_lc[] = {" ", "\n", /*candidate content*/
                                                    "1#", "q", "w", "e", "r", "t", "y", "u", "i", "o", "p", LV_SYMBOL_BACKSPACE, "\n",
                                                    "ABC", "a", "s", "d", "f", "g", "h", "j", "k", "l", LV_SYMBOL_NEW_LINE, "\n",
                                                    "_", "-", "z", "x", "c", "v", "b", "n", "m", ".", ",", ":", "\n",
                                                    LV_SYMBOL_KEYBOARD,  CHINESE_KEY PEERSIAN_KEY  LV_SYMBOL_LEFT, " ", LV_SYMBOL_RIGHT, LV_SYMBOL_OK, ""
                                                   };

```

Reply:
> It would be more elegant to have something like:
> 
> ```c
> #if LV_USE_ARABIC_PERSIAN_CHARS 
>   #define PEERSIAN_KEY  "أب",
>   #define PEERSIAN_CTRL  LV_KEYBOARD_CTRL_BUTTON_FLAGS | 2,
> #else
>   #define PEERSIAN_KEY  
>   #define PEERSIAN_CTRL
> #endif
> //Same for Chines
> //..
> 
> static const char * const default_en_kb_map_lc[] = {" ", "\n", /*candidate content*/
>                                                     "1#", "q", "w", "e", "r", "t", "y", "u", "i", "o", "p", LV_SYMBOL_BACKSPACE, "\n",
>                                                     "ABC", "a", "s", "d", "f", "g", "h", "j", "k", "l", LV_SYMBOL_NEW_LINE, "\n",
>                                                     "_", "-", "z", "x", "c", "v", "b", "n", "m", ".", ",", ":", "\n",
>                                                     LV_SYMBOL_KEYBOARD,  CHINESE_KEY PEERSIAN_KEY  LV_SYMBOL_LEFT, " ", LV_SYMBOL_RIGHT, LV_SYMBOL_OK, ""
>                                                    };
> ```

okay, i will try it